src/scheduler/server_info.cpp

```diff
@@ -286,15 +282,14 @@ query_server(status *pol, int pbs_sd)
 	 * in the case we don't sort the jobs and don't have by_queue turned on
 	 */
 	if ((policy->round_robin == 1) || (policy->by_queue == 1))
-		qsort(sinfo->queues, sinfo->num_queues, sizeof(queue_info *),
-			cmp_queue_prio_dsc);
+		std::sort(sinfo->queues.begin(), sinfo->queues.end(), cmp_queue_prio_dsc);
 	if (policy->round_robin == 1) {
 		/* queues are already sorted in descending order of their priority */
-		for (i = 0; i < sinfo->num_queues; i++) {
-			auto ret_val = add_queue_to_list(&sinfo->queue_list, sinfo->queues[i]);
+		for (auto queue : sinfo->queues) {
```

Comment:
I'll change the variable name, I can't combine the loops because this loop has to happen only in the case of round_robin policy and that too after the queues have been sorted.